lib/main_transactional.pm

```diff
@@ -0,0 +1,283 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: module loader for Transactional product family:
+#  ALP
+#  MicroOS
+#  SLE-Micro
+#  Leap-Micro
+# Maintainer: qa-c@suse.de
+
+package main_transactional;
+use strict;
+use warnings;
+use base 'Exporter';
+use Exporter;
+use main_common;
+use main_containers qw(load_container_tests is_container_test);
+use testapi qw(check_var get_required_var get_var set_var);
+use version_utils qw(is_microos is_sle_micro is_leap_micro is_alp is_staging is_released is_transactional is_rt);
+use utils;
+use Utils::Architectures;
+use Utils::Backends;
+
+our @EXPORT = qw(
+  load_tests
+);
+
+sub is_image {
+    return get_required_var('FLAVOR') =~ /image|default|kvm/i;
+}
+
+sub is_dvd {
+    return get_required_var('FLAVOR') =~ /dvd/i;
+}
+
+sub is_regproxy_required {
+    # For now only the Kubic kubeadm test needs a registry proxy.
+    # docker_image and podman_image pull with the full path themselves.
+    return check_var('SYSTEM_ROLE', 'kubeadm');
+}
+
+sub load_config_tests {
+    if (is_microos) {
+        loadtest 'update/zypper_clear_repos';
+        loadtest 'console/zypper_ar';
+        loadtest 'console/zypper_ref';
+    }
+    loadtest 'transactional/tdup' if get_var('TDUP');
+    loadtest 'transactional/host_config' unless is_dvd;
+    loadtest 'rt/rt_is_realtime' if is_rt;
+    loadtest 'transactional/enable_selinux' if (get_var('ENABLE_SELINUX') && is_image);
+    loadtest 'console/suseconnect_scc' if (is_sle_micro && get_var('SCC_REGISTER') && !is_dvd);
+    loadtest 'transactional/install_updates' if (is_sle_micro && is_released);
+}
+
+sub load_boot_from_disk_tests {
+    if (is_s390x) {
+        loadtest 'installation/bootloader_start';
+        loadtest 'boot/boot_to_desktop';
+    } else {
+        if (check_var('FIRST_BOOT_CONFIG', 'wizard')) {
+            loadtest 'jeos/firstrun';
+        } else {
+            loadtest 'microos/disk_boot';
+        }
+    }
+    loadtest 'installation/system_workarounds' if (is_aarch64 && is_microos);
+    replace_opensuse_repos_tests if is_repo_replacement_required;
+}
+
+sub load_boot_from_dvd_tests {
+    if (is_s390x) {
+        loadtest 'installation/bootloader_start';
+    } elsif (get_var("UEFI")) {
+        loadtest 'installation/bootloader_uefi';
```

Comment:
I would rather prefer to make this one default. For instance, it can handle `EXTRABOOTPARAMS` which can be useful to inject some grub parameters. And the module works fine for x86_64 uefi, bios or aarch64

Reply:
We do it in JeOS and it works fine. :)

Reply:
bad idea :(
https://openqa.suse.de/tests/10082654